ui-packages/packages/trusty/src/types.ts

```diff
@@ -133,8 +135,15 @@ export interface CFSearchInput extends ItemObject {
   domain?: CFNumericalDomain | CFCategoricalDomain;
 }
 
+export enum CFGoalRole {
+  UNSUPPORTED,
+  ORIGINAL,
+  FIXED,
+  FLOATING
+}
+
 export type CFGoal = Pick<ItemObject, 'name' | 'kind' | 'typeRef' | 'value'> & {
-  isFixed: boolean;
```

Comment:
I changed `isFixed` to a more generic "role" (since a CF can only be one or another): `FIXED`, `FLOATING` (i.e. don't care), `ORIGINAL` and `UNSUPPORTED` (i.e. structures).